src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -2963,43 +2963,269 @@ mesh::topology::unstructured::generate_sides(const Node &topo,
+    volume_dependent_helper(const Node &topo_dest,
+                            const Node &coordset_dest,
+                            const std::string coordset_name,
+                            const int dimensions,
+                            const int num_new_shapes, // number of new triangles or tetrahedrons
+                            const int num_orig_shapes, // number of original polygons or polyhedra
+                            const T *tri_to_poly) 
+    {
+
+        // TODO we want to make volume into a field
+
+        Node *volumes = new Node;
+
+        // first we calculate the volume of each triangle
+        (*volumes)["tri"].set(conduit::DataType::float64(num_new_shapes));
+        float64 *tri_volumes = (*volumes)["tri"].value();
+
+        const U *connec = topo_dest["elements/connectivity"].value();
+        const V *coords_x = coordset_dest[coordset_name + "/values/x"].value();
+        const V *coords_y = coordset_dest[coordset_name + "/values/y"].value();
+
+        if (dimensions == 2)
+        {
+            for (int i = 0; i < num_new_shapes; i ++)
+            {
+                float x1 = coords_x[connec[i * 3 + 0]];
+                float y1 = coords_y[connec[i * 3 + 0]];
+                float x2 = coords_x[connec[i * 3 + 1]];
+                float y2 = coords_y[connec[i * 3 + 1]];
+                float x3 = coords_x[connec[i * 3 + 2]];
+                float y3 = coords_y[connec[i * 3 + 2]];
+
+                tri_volumes[i] = triangle_area(x1, y1, x2, y2, x3, y3);
+            }
+        }
+        else if (dimensions == 3)
+        {
+            const V *coords_z = coordset_dest[coordset_name + "/values/z"].value();
+
+            for (int i = 0; i < num_new_shapes; i ++)
+            {
+                float x1 = coords_x[connec[i * 4 + 0]];
+                float y1 = coords_y[connec[i * 4 + 0]];
+                float z1 = coords_z[connec[i * 4 + 0]];
+                float x2 = coords_x[connec[i * 4 + 1]];
+                float y2 = coords_y[connec[i * 4 + 1]];
+                float z2 = coords_z[connec[i * 4 + 1]];
+                float x3 = coords_x[connec[i * 4 + 2]];
+                float y3 = coords_y[connec[i * 4 + 2]];
+                float z3 = coords_z[connec[i * 4 + 2]];
+                float x4 = coords_x[connec[i * 4 + 3]];
+                float y4 = coords_y[connec[i * 4 + 3]];
+                float z4 = coords_z[connec[i * 4 + 3]];
+
+                tri_volumes[i] = tetrahedron_volume(x1, y1, z1, x2, y2, z2, x3, y3, z3, x4, y4, z4);
+            }
+        }
+        else
+        {
+            CONDUIT_ERROR("Higher dimensions are not supported.");
+        }
+
+        // next we calculate the volume of each polygon
+        (*volumes)["poly"].set(conduit::DataType::float64(num_orig_shapes));
+        float64 *poly_volumes = (*volumes)["poly"].value();
+
+        for (int i = 0; i < num_orig_shapes; i ++)
+        {
+            poly_volumes[i] = 0;
```

Comment:
Can I assume with `set()` that everything will be set to 0, or should I do this for loop regardless?